[Bug]: test test_object_storage_encryption_configuration_dashboard failing due TimeoutException

### Description
```
2026-04-04 12:37:57 - _locator = ("//button[contains(@class, 'c-button') and contains(@class, 'm-link') and contains(@class, 'm-inline') and (text()='Enabled' or text()='Not enabled')]", 'xpath')
_timeout = 30, _enable_screenshot = True, _copy_dom = False

    def _do_click(_locator, _timeout=30, _enable_screenshot=False, _copy_dom=False):
        # wait for page fully loaded only if an element was not located
        # prevents needless waiting and frequent crushes on ODF Overview page,
        # when metrics and alerts frequently updated
        if not self.get_elements(_locator):
            self.page_has_loaded()
        screenshot = (
            ocsci_config.UI_SELENIUM.get("screenshot") and enable_screenshot
        )
        date_time = datetime.datetime.now().strftime("%Y-%m-%dT%H-%M-%S.%f")
        if screenshot:
            self.take_screenshot(f"{type(self).__name__}-{date_time}")
        if _copy_dom:
            self.copy_dom(f"{type(self).__name__}-{date_time}")
    
        wait = WebDriverWait(self.driver, timeout)
        try:
>           element = wait.until(
                ec.element_to_be_clickable((locator[1], locator[0]))
            )

ocs_ci/ocs/ui/base_ui.py:238: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <selenium.webdriver.support.wait.WebDriverWait (session="7093d03239e298bdac07d435135d63fb")>
method = <function element_to_be_clickable.<locals>._predicate at 0x7ff28f494cc0>
message = ''

    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = "") -> T:
        """Wait until the method returns a value that is not False.
    
        Calls the method provided with the driver as an argument until the
        return value does not evaluate to ``False``.
    
        Parameters:
        -----------
        method: callable(WebDriver)
            - A callable object that takes a WebDriver instance as an argument.
    
        message: str
            - Optional message for :exc:`TimeoutException`
    
        Return:
        -------
        object: T
            - The result of the last call to `method`
    
        Raises:
        -------
        TimeoutException
            - If 'method' does not return a truthy value within the WebDriverWait
            object's timeout
    
        Example:
        --------
        >>> from selenium.webdriver.common.by import By
        >>> from selenium.webdriver.support.ui import WebDriverWait
        >>> from selenium.webdriver.support import expected_conditions as EC
    
        # Wait until an element is visible on the page
        >>> wait = WebDriverWait(driver, 10)
        >>> element = wait.until(EC.visibility_of_element_located((By.ID, "exampleId")))
        >>> print(element.text)
        """
        screen = None
        stacktrace = None
    
        end_time = time.monotonic() + self._timeout
        while True:
            try:
                value = method(self._driver)
                if value:
                    return value
            except self._ignored_exceptions as exc:
                screen = getattr(exc, "screen", None)
                stacktrace = getattr(exc, "stacktrace", None)
            if time.monotonic() > end_time:
                break
            time.sleep(self._poll)
>       raise TimeoutException(message, screen, stacktrace)
E       selenium.common.exceptions.TimeoutException: Message: 
E       Stacktrace:
E       #0 0x56543992471a <unknown>
E       #1 0x5654393c70a0 <unknown>
E       #2 0x5654394189b0 <unknown>
E       #3 0x565439418ba1 <unknown>
E       #4 0x565439466ea4 <unknown>
E       #5 0x56543943e3cd <unknown>
E       #6 0x5654394642a0 <unknown>
E       #7 0x56543943e173 <unknown>
E       #8 0x56543940ad4b <unknown>
E       #9 0x56543940b9b1 <unknown>
E       #10 0x5654398e98cb <unknown>
E       #11 0x5654398ed7ca <unknown>
E       #12 0x5654398d1622 <unknown>
E       #13 0x5654398ee354 <unknown>
E       #14 0x5654398b645f <unknown>
E       #15 0x5654399124f8 <unknown>
E       #16 0x5654399126d6 <unknown>
E       #17 0x565439923586 <unknown>
E       #18 0x7f3e6ec8b2fa start_thread

venv/lib64/python3.11/site-packages/selenium/webdriver/support/wait.py:138: TimeoutException

During handling of the above exception, another exception occurred:

self = <tests.functional.encryption.test_encryption_configuration_dashboard.TestEncryptionConfigurationDashboard object at 0x7ff2887d3cd0>
setup_ui_class = <selenium.webdriver.chrome.webdriver.WebDriver (session="7093d03239e298bdac07d435135d63fb")>

    @pytest.mark.polarion_id("OCS-6301")
    def test_object_storage_encryption_configuration_dashboard(self, setup_ui_class):
        """Test the encryption configuration dashboard of Object details for correctness.
    
        Steps:
            1. Navigate to object details page
            2. Open encryption details.
            3. verify encryption data with the nooba and storagecluster spec.
        """
        # Navigate to the Object Storage page
        object_details_page = PageNavigator().navigate_object_tab()
    
>       encryption_summary = object_details_page.get_object_encryption_summary()

tests/functional/encryption/test_encryption_configuration_dashboard.py:142: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ocs_ci/ocs/ui/page_objects/encryption_module.py:102: in get_object_encryption_summary
    return self._get_encryption_summary("object_storage")
ocs_ci/ocs/ui/page_objects/encryption_module.py:31: in _get_encryption_summary
    self.do_click(
ocs_ci/ocs/ui/base_ui.py:258: in do_click
    _do_click(locator, timeout, enable_screenshot, copy_dom)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

_locator = ("//button[contains(@class, 'c-button') and contains(@class, 'm-link') and contains(@class, 'm-inline') and (text()='Enabled' or text()='Not enabled')]", 'xpath')
_timeout = 30, _enable_screenshot = True, _copy_dom = False

    def _do_click(_locator, _timeout=30, _enable_screenshot=False, _copy_dom=False):
        # wait for page fully loaded only if an element was not located
        # prevents needless waiting and frequent crushes on ODF Overview page,
        # when metrics and alerts frequently updated
        if not self.get_elements(_locator):
            self.page_has_loaded()
        screenshot = (
            ocsci_config.UI_SELENIUM.get("screenshot") and enable_screenshot
        )
        date_time = datetime.datetime.now().strftime("%Y-%m-%dT%H-%M-%S.%f")
        if screenshot:
            self.take_screenshot(f"{type(self).__name__}-{date_time}")
        if _copy_dom:
            self.copy_dom(f"{type(self).__name__}-{date_time}")
    
        wait = WebDriverWait(self.driver, timeout)
        try:
            element = wait.until(
                ec.element_to_be_clickable((locator[1], locator[0]))
            )
            element.click()
        except TimeoutException as e:
            self.take_screenshot(f"{type(self).__name__}-{date_time}")
            self.copy_dom(f"{type(self).__name__}-{date_time}")
            logger.error(e)
            new_locator = self.locator_fallback.attempt_fallback(locator, "click")
            if new_locator:
                element = WebDriverWait(self.driver, timeout).until(
                    ec.element_to_be_clickable((new_locator[1], new_locator[0]))
                )
                element.click()
                return
>           raise TimeoutException(
                f"Failed to find the element ({locator[1]},{locator[0]})"
            )
E           selenium.common.exceptions.TimeoutException: Message: Failed to find the element (xpath,//button[contains(@class, 'c-button') and contains(@class, 'm-link') and contains(@class, 'm-inline') and (text()='Enabled' or text()='Not enabled')])

ocs_ci/ocs/ui/base_ui.py:253: TimeoutException
```

### Steps to Reproduce
1.
2.
3.

### Actual Behavior

### Expected Behavior

### Impact (likelihood of reproduction, impact on the cluster and on other tests, etc.)

### Screenshots (if applicable)

### Environment
- Test Suite(s):
- Platform(s):
- Version(s):
- OS:

### Additional Context

--- a/ocs_ci/ocs/constants.py
+++ b/ocs_ci/ocs/constants.py
@@ -3788,6 +3788,8 @@ ENCRYPTION_DASHBOARD_CONTEXT_MAP = {
     "In-transit encryption": "intransit_encryption",
     "Object storage": "object_storage",
 }
+# PatternFly success/enabled icon colors (PF5 color/fill attrs; PF6 prefers data-test)
+UI_SUCCESS_ICON_COLORS = ("#3e8635", "#1a7f37", "#3d7317")
 
 # Multiple device classes Yaml files
 STORAGE_DEVICESET_YAML = os.path.join(

--- a/ocs_ci/ocs/ui/page_objects/data_foundation_tabs_common.py
+++ b/ocs_ci/ocs/ui/page_objects/data_foundation_tabs_common.py
@@ -607,7 +607,10 @@ class DataFoundationTabBar(PageNavigator):
         Navigate to Block and File tab. Accessible from any Data Foundation tabs
         """
         self.do_click(self.validation_loc["block-and-file-tab"], enable_screenshot=True)
-        self.page_has_loaded()
+        self.wait_for_element_to_be_present(
+            self.validation_loc["block-and-file-tab-active"], timeout=60
+        )
+        self.wait_for_encryption_summary_ready("file_and_block")
 
         from ocs_ci.ocs.ui.page_objects.block_and_file import BlockAndFile
 
@@ -629,7 +632,10 @@ class DataFoundationTabBar(PageNavigator):
         Navigate to Object tab. Accessible from any Data Foundation tabs
         """
         self.do_click(self.validation_loc["object-tab"], enable_screenshot=True)
-        self.page_has_loaded()
+        self.wait_for_element_to_be_present(
+            self.validation_loc["object-tab-active"], timeout=60
+        )
+        self.wait_for_encryption_summary_ready("object_storage")
 
         from ocs_ci.ocs.ui.page_objects.object_storage import ObjectStorage
 

--- a/ocs_ci/ocs/ui/page_objects/encryption_module.py
+++ b/ocs_ci/ocs/ui/page_objects/encryption_module.py
@@ -1,7 +1,11 @@
+from selenium.common.exceptions import NoSuchElementException
 from selenium.webdriver.common.by import By
 
 from ocs_ci.ocs.ui.helpers_ui import logger
-from ocs_ci.ocs.constants import ENCRYPTION_DASHBOARD_CONTEXT_MAP
+from ocs_ci.ocs.constants import (
+    ENCRYPTION_DASHBOARD_CONTEXT_MAP,
+    UI_SUCCESS_ICON_COLORS,
+)
 from ocs_ci.ocs.ui.page_objects.data_foundation_tabs_common import (
     DataFoundationTabBar,
 )
@@ -27,11 +31,14 @@ class EncryptionModule(DataFoundationTabBar):
 
         logger.info(f"Getting Encryption Summary for context: {context_key}")
 
+        self.wait_for_encryption_summary_ready(context_key)
+
+        enabled_locator = self.validation_loc["encryption_summary"][context_key][
+            "enabled"
+        ]
+
         # Open the encryption summary popup
-        self.do_click(
-            self.validation_loc["encryption_summary"][context_key]["enabled"],
-            enable_screenshot=True,
-        )
+        self.do_click(enabled_locator, enable_screenshot=True)
 
         self.page_has_loaded(
             module_loc=self.validation_loc["encryption_summary"][context_key][
@@ -49,22 +56,14 @@ class EncryptionModule(DataFoundationTabBar):
             raise ValueError("Error getting root web element")
         root_element = root_elements[0]
 
-        # Extract headers and statuses
-        enc_headers = [
-            head
-            for head in root_element.find_elements(By.TAG_NAME, "h6")
-            if head.text in ENCRYPTION_DASHBOARD_CONTEXT_MAP
-        ]
-        enc_status = [
-            svg
-            for svg in root_element.find_elements(By.TAG_NAME, "svg")
-            if svg.get_attribute("color")
-        ]
-
-        for header, svg in zip(enc_headers, enc_status):
-            context = ENCRYPTION_DASHBOARD_CONTEXT_MAP[header.text]
-            encryption_summary[context]["status"] = (
-                svg.get_attribute("color") == "#3e8635"
+        # Extract headers and statuses (PF6 uses data-test on icons, not color attr)
+        for header in root_element.find_elements(By.TAG_NAME, "h6"):
+            header_text = EncryptionModule._encryption_header_text(header)
+            if header_text not in ENCRYPTION_DASHBOARD_CONTEXT_MAP:
+                continue
+            context = ENCRYPTION_DASHBOARD_CONTEXT_MAP[header_text]
+            encryption_summary[context]["status"] = self._encryption_row_status_enabled(
+                header
             )
 
         # Process encryption summary text
@@ -91,6 +90,63 @@ class EncryptionModule(DataFoundationTabBar):
         )
 
         return encryption_summary
+
+    @staticmethod
+    def _encryption_header_text(header_element) -> str:
+        """Return visible header label (PF6 h6 elements may have empty .text)."""
+        text = (header_element.text or "").strip()
+        if text:
+            return text
+        return (header_element.get_attribute("textContent") or "").strip()
+
+    def _encryption_row_status_enabled(self, header_element) -> bool:
+        """
+        Return whether the encryption row shows an enabled/success status icon.
+
+        PF5 icons expose color="#3e8635"; PF6 icons use data-test="success-icon".
+        """
+        row = None
+        for row_xpath in self.validation_loc["encryption_summary"]["status_row_xpaths"]:
+            try:
+                row = header_element.find_element(By.XPATH, row_xpath)
+                break
+            except NoSuchElementException:
+                continue
+            except Exception as e:
+                logger.warning(
+                    "Unexpected error locating encryption status row via %s: %s",
+                    row_xpath,
+                    e,
+                )
+                raise
+
+        if row is None:
+            return False
+
+        if row.find_elements(By.CSS_SELECTOR, '[data-test="success-icon"]'):
+            return True
+
+        for svg in row.find_elements(By.TAG_NAME, "svg"):
+            color = (
+                svg.get_attribute("color") or svg.get_attribute("fill") or ""
+            ).lower()
+            if color in UI_SUCCESS_ICON_COLORS:
+                return True
+        return False
+
+    def wait_for_encryption_summary_ready(self, context_key):
+        """
+        Wait until the encryption summary control is present on the dashboard.
+
+        Args:
+            context_key (str): Key under encryption_summary locators
+                (e.g. "file_and_block", "object_storage").
+        """
+        enabled_locator = self.validation_loc["encryption_summary"][context_key][
+            "enabled"
+        ]
+        self.wait_for_element_to_be_present(enabled_locator, timeout=60)
+        self.page_has_loaded(module_loc=enabled_locator, retries=15, sleep_time=2)
 
     def get_object_encryption_summary(self):
         """

--- a/ocs_ci/ocs/ui/page_objects/page_navigator.py
+++ b/ocs_ci/ocs/ui/page_objects/page_navigator.py
@@ -105,13 +105,20 @@ class PageNavigator(BaseUI):
         self.choose_expanded_mode(mode=True, locator=self.page_nav["Storage"])
         self.do_click(locator=self.page_nav["storage_cluster"], timeout=90)
         self.page_has_loaded(retries=15)
+
+        from ocs_ci.ocs.ui.page_objects.storage_cluster import StorageClusterPage
+
+        storage_cluster_page = StorageClusterPage()
+        storage_cluster_page.wait_for_element_to_be_present(
+            storage_cluster_page.validation_loc["block-and-file-tab-active"],
+            timeout=60,
+        )
+        storage_cluster_page.wait_for_encryption_summary_ready("file_and_block")
         logger.info(f"Default page is {self.driver.title}")
 
         logger.info("Successfully navigated to ODF tab under Storage section")
 
-        from ocs_ci.ocs.ui.page_objects.storage_cluster import StorageClusterPage
-
-        return StorageClusterPage()
+        return storage_cluster_page
 
     def nav_object_storage_page(self):
         """

--- a/ocs_ci/ocs/ui/page_objects/storage_cluster.py
+++ b/ocs_ci/ocs/ui/page_objects/storage_cluster.py
@@ -24,17 +24,20 @@ class StorageClusterPage(
 
     def validate_block_and_file_tab_active(self) -> bool:
         """
-        Validate Overview tab is active
+        Validate Block and File tab is active and encryption summary is loaded.
 
         Returns:
             bool: True if active, False otherwise
         """
         logger.info("Validate Block and File tab is active")
-        is_default = self.is_block_and_file_tab()
-        if not is_default:
-            logger.warning("Block and File tab is not active")
+        if not self.is_block_and_file_tab():
+            logger.warning("Block and File tab is not active, navigating to it")
+            self.nav_block_and_file_tab()
 
-        return is_default
+        is_active = self.is_block_and_file_tab()
+        if is_active:
+            self.wait_for_encryption_summary_ready("file_and_block")
+        return is_active
 
     def nav_cephblockpool_verify_statusready(self):
         """

--- a/ocs_ci/ocs/ui/views.py
+++ b/ocs_ci/ocs/ui/views.py
@@ -2375,38 +2375,40 @@ validation_4_17 = {
     ),
 }
 
+_ENCRYPTION_SUMMARY_ENABLED = (
+    "//dt[@data-test='detail-item-title' and normalize-space()='Encryption']"
+    "/following-sibling::dd[@data-test='detail-item-value']"
+    "//button[normalize-space()='Enabled' or normalize-space()='Not enabled']",
+    By.XPATH,
+)
+_ENCRYPTION_SUMMARY_CLOSE = (
+    "//div[@role='dialog']//button[@aria-label='Close']",
+    By.XPATH,
+)
+_ENCRYPTION_SUMMARY_CONTENT = (
+    "//div[@role='dialog'][.//button[@aria-label='Close']][.//h6]",
+    By.XPATH,
+)
+# Relative xpaths from an encryption summary h6 to its status row (PF5/PF6)
+_ENCRYPTION_STATUS_ROW_XPATHS = (
+    "./parent::*/parent::*",
+    "./ancestor::*[contains(@class, 'l-flex') and not(contains(@class, '__item'))][1]",
+    "./ancestor::*[contains(@class, 'flex') and not(contains(@class, '__item'))][1]",
+)
+
 validation_4_18 = {
     "encryption_summary": {
         "file_and_block": {
-            "enabled": (
-                "//button[contains(@class, 'c-button') and contains(@class, 'm-link') and contains(@class, 'm-inline') "
-                "and (text()='Enabled' or text()='Not enabled')]",
-                By.XPATH,
-            ),
-            "close": (
-                "//button[contains(@class, 'c-button') and contains(@class, 'm-plain') and @aria-label='Close']",
-                By.XPATH,
-            ),
-            "encryption_content_data": (
-                "//div[contains(@class, 'c-popover__body')]",
-                By.XPATH,
-            ),
+            "enabled": _ENCRYPTION_SUMMARY_ENABLED,
+            "close": _ENCRYPTION_SUMMARY_CLOSE,
+            "encryption_content_data": _ENCRYPTION_SUMMARY_CONTENT,
         },
         "object_storage": {
-            "enabled": (
-                "//button[contains(@class, 'c-button') and contains(@class, 'm-link') and contains(@class, 'm-inline') "
-                "and (text()='Enabled' or text()='Not enabled')]",
-                By.XPATH,
-            ),
-            "close": (
-                "//button[contains(@class, 'c-button') and contains(@class, 'm-plain') and @aria-label='Close']",
-                By.XPATH,
-            ),
-            "encryption_content_data": (
-                "//div[contains(@class, 'c-popover__content')]",
-                By.XPATH,
-            ),
+            "enabled": _ENCRYPTION_SUMMARY_ENABLED,
+            "close": _ENCRYPTION_SUMMARY_CLOSE,
+            "encryption_content_data": _ENCRYPTION_SUMMARY_CONTENT,
         },
+        "status_row_xpaths": _ENCRYPTION_STATUS_ROW_XPATHS,
     }
 }
 validation_4_20 = {
@@ -2414,8 +2416,15 @@ validation_4_20 = {
         "//button[@data-test='horizontal-link-Block and File' and @aria-selected='true']",
         By.XPATH,
     ),
-    "block-and-file-tab": ("//span[contains(.,'Block and File')]", By.XPATH),
+    "block-and-file-tab": (
+        "//button[@data-test='horizontal-link-Block and File']",
+        By.XPATH,
+    ),
     "object-tab": ("//button[@data-test='horizontal-link-Object']", By.XPATH),
+    "object-tab-active": (
+        "//button[@data-test='horizontal-link-Object' and @aria-selected='true']",
+        By.XPATH,
+    ),
     "storage-pools-tab": (
         "//button[@data-test='horizontal-link-Storage pools']",
         By.XPATH,

--- a/tests/functional/encryption/test_encryption_configuration_dashboard.py
+++ b/tests/functional/encryption/test_encryption_configuration_dashboard.py
@@ -2,7 +2,11 @@ import pytest
 import logging
 
 log = logging.getLogger(__name__)
-from ocs_ci.framework.pytest_customization.marks import green_squad, tier1
+from ocs_ci.framework.pytest_customization.marks import (
+    green_squad,
+    skipif_external_mode,
+    tier1,
+)
 from ocs_ci.ocs.ocp import OCP
 from ocs_ci.framework import config
 from ocs_ci.ocs import constants
@@ -70,6 +74,7 @@ class TestEncryptionConfigurationDashboard:
         assert actual_status == expected_status, error_message
         log.info(f"{context} status is as expected: {actual_status}")
 
+    @skipif_external_mode
     @pytest.mark.polarion_id("OCS-6300")
     def test_file_block_encryption_configuration_dashboard(self, setup_ui_class):
         """Test the encryption configuration dashboard of File And Block details for correctness.
@@ -127,6 +132,7 @@ class TestEncryptionConfigurationDashboard:
             "InTransit Encryption status is incorrect in the dashboard.",
         )
 
+    @skipif_external_mode
     @pytest.mark.polarion_id("OCS-6301")
     def test_object_storage_encryption_configuration_dashboard(self, setup_ui_class):
         """Test the encryption configuration dashboard of Object details for correctness.

